src/HTTPure/Body.purs

```diff
@@ -17,45 +17,67 @@ import Node.Encoding as Encoding
 import Node.HTTP as HTTP
 import Node.Stream as Stream
 
+import HTTPure.Headers as Headers
+
+newtype Chunked = Chunked (Stream.Readable ())
+
 -- | Types that implement the `Body` class can be used as a body to an HTTPure
 -- | response, and can be used with all the response helpers.
 class Body b where
 
-  -- | Given a body value, return an effect that maybe calculates a size.
-  -- | TODO: This is a `Maybe` to support chunked transfer encoding.  We still
-  -- | need to add code to send the body using chunking if the effect resolves a
-  -- | `Maybe.Nothing`.
-  size :: b -> Effect.Effect (Maybe.Maybe Int)
+  -- | Return any additional headers that need to be sent with this body type.
+  -- | Things like `Content-Type`, `Content-Length`, and `Transfer-Encoding`.
+  additionalHeaders :: b -> Effect.Effect Headers.Headers
```

Comment:
It would be a nice addition to document somewhere whether the user can override these additional headers by using e.g. `ok'`, or are they applied after user defined headers.

Reply:
Can do; in fact, they cannot be overridden with the current implementation but I should reverse that.  For now, I'll document it here; I have a ticket open to revisit the guides (#106) before we release the next version and I'll make sure to add a note there about overriding default headers